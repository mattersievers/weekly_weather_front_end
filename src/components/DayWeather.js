function DayWeather(props){
    let {daily} = props.weatherData;    

    //Gives the string for the day of the week when input Unix time from the weather API
    let dayOfWeek = (day) => {
        const weekdays = ['Sunday','Monday','Tuesday','Wednesday','Thursday','Friday','Saturday']
        var dayNum = new Date(day.dt * 1000).getDay();
        return weekdays[dayNum];
    }

    if(!props){
        return (
            <div>Rendering...</div>
        )
    }
    console.log(daily)
    return (
        <div>
            <div className="day">{dayOfWeek(daily[0])}</div>
            <img src={`http://openweathermap.org/img/wn/${daily[0].weather[0].icon}@2x.png`} />
            <div className="max">{parseInt(daily[0].temp.max)} °  <span className="min">{parseInt(daily[0].temp.min)} ° </span></div>
        </div>
    )
}

export default DayWeather;
