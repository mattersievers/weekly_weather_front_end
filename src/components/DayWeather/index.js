function DayWeather(props){
    let {daily} = props.weatherData;    
    let {dayNumber} = props;

    //Gives the string for the day of the week when input Unix time from the weather API
    let dayOfWeek = (day) => {
        const weekdays = ['Sun','Mon','Tues','Wed','Thurs','Fri','Sat']
        var dayNum = new Date(day.dt * 1000).getDay();
        return weekdays[dayNum];
    }

    if(!props){
        return (
            <div>Rendering...</div>
        )
    }

    return (
        <div className="dayContainer">
            <div className="day" data-testid="dayOfWeek">{dayOfWeek(daily[dayNumber])}</div>
            <img data-testid="weatherImage" src={`http://openweathermap.org/img/wn/${daily[dayNumber].weather[0].icon}@2x.png`} alt={daily[dayNumber].weather[0].description}/>
            <div className="maxTemp" data-testid="maxTemp">{parseInt(daily[dayNumber].temp.max)}°  <span className="minTemp" data-testid="minTemp">{parseInt(daily[dayNumber].temp.min)}° </span></div>
        </div>
    )
}

export default DayWeather;
