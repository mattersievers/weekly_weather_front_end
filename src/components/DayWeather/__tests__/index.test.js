import React from "react";
import { render, cleanup } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import DayWeather from "../index";

const weatherData = require('./api_test.json');
const dayNumber = 3;

afterEach(cleanup);

describe('Day Weather component', () => {
    it('renders', () =>{
        render(<DayWeather 
                weatherData = {weatherData}
                dayNumber = {dayNumber}
                />);
    });

    it('matches snapshot DOM node structure', () => {
        const {asFragment} = render(<DayWeather 
                                    weatherData = {weatherData}
                                    dayNumber = {dayNumber}
                                    />);
        expect (asFragment()).toMatchSnapshot();
    })

    it('day of the week is Friday using api_test.json', () => {
        const { getByTestId } = render(<DayWeather 
                                    weatherData = {weatherData}
                                    dayNumber = {dayNumber}
                                    />);
        expect (getByTestId('dayOfWeek')).toHaveTextContent('Fri');
    })

    it('day of the week is Friday and icon and temps load according data from api_test.json', () => {
        const { getByTestId } = render(<DayWeather 
                                    weatherData = {weatherData}
                                    dayNumber = {dayNumber}
                                    />);
        expect (getByTestId('dayOfWeek')).toHaveTextContent('Fri');
        expect (getByTestId('weatherImage')).toHaveProperty('src','http://openweathermap.org/img/wn/04d@2x.png');
        expect (getByTestId('weatherImage')).toHaveProperty('alt','overcast clouds');
        expect (getByTestId('maxTemp')).toHaveTextContent('85° 68°');
        expect (getByTestId('minTemp')).toHaveTextContent('68°');
    })
})