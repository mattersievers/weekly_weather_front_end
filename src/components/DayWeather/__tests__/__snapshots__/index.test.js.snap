// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day Weather component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="dayContainer"
  >
    <div
      class="day"
      data-testid="dayOfWeek"
    >
      Fri
    </div>
    <img
      alt="overcast clouds"
      data-testid="weatherImage"
      src="http://openweathermap.org/img/wn/04d@2x.png"
    />
    <div
      class="maxTemp"
      data-testid="maxTemp"
    >
      85°  
      <span
        class="minTemp"
        data-testid="minTemp"
      >
        68° 
      </span>
    </div>
  </div>
</DocumentFragment>
`;
